### PR TITLE
Updates the codebase to the latest kcp version and dependencies

### PR DESCRIPTION
**Pull request by jmprusi**
## Description of Changes

This PR updates the codebase to the latest kcp version and dependencies.

Some of the changes include:
* Update the code-generation
* Changes to how golang tools are installed
* logicalcluster v2->v3
* kcp and kcp-kubernetes pinned to the latest available version
* kcp.dev -> kcp.io renaming of resources
* New placement for cert-manager

### CODE REVIEW

**Comment by maleck13** [comment]
Happy to help out with reviews. I took a look through what is there now but seems to largely be bringing the code base upto date. Changes look ok so far

File: docs/observability/generated_metrics.adoc [hunk 1]
```diff
@@ -112,8 +84,6 @@ and the 2nd column is the table title (e.g. "Go Runtime Metrics")
 | `go_memstats_stack_inuse_bytes` | Number of bytes in use by the stack allocator.| GAUGE| 
 | `go_memstats_stack_sys_bytes` | Number of bytes obtained from system for stack allocator.| GAUGE| 
 | `go_memstats_sys_bytes` | Number of bytes obtained from system.| GAUGE| 
-| `go_sched_goroutines_goroutines` | Count of live goroutines.| GAUGE| 
-| `go_sched_latencies_seconds` | Distribution of the time goroutines have spent in the scheduler in a runnable state before actually running.| HISTOGRAM| 
 | `go_threads` | Number of OS threads created.| GAUGE| 
 |===
 .Process metrics
```
[the remark this answers is not in the document]
Comment (jmprusi) [reply docs/observability/generated_metrics.adoc:89]: that's what I did, generated the metrics from the prometheus output, I'm not sure why those changed, but looks like most of those metrics are golang version dependant.

File: config/apiexports/kustomization.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 resources:
+  - ./apiexport/
   - ./apibinding/
-  - ./apiexport/
\ No newline at end of file
+  # - ./rbac/
```
[the remark this answers is not in the document]
Comment (jmprusi) [reply config/apiexports/kustomization.yaml:4]: leftover trying to debug some issue, I can remove those.

**Pull request merged**